csharp/extractor/Semmle.Extraction.CSharp/Entities/OrdinaryMethod.cs

```diff
@@ -47,6 +52,18 @@ public override void Populate(TextWriter trapFile)
             ExtractCompilerGenerated(trapFile);
         }
 
+        private bool IsCompilerGeneratedDelegate()
+        {
+            // Lambdas with parameter defaults or a `params` parameter are implemented
+            // using compiler generated delegate types.
+            if (Symbol.MethodKind == MethodKind.DelegateInvoke &&
+                Symbol.ContainingType is INamedTypeSymbol nt)
+            {
+                return nt.TypeKind == TypeKind.Delegate && nt.IsImplicitlyDeclared;
+            }
```

Comment:
Is the `Symbol` itself `IsImplicitlyDeclared` for the cases that we're interested in? There's probably no point in checking the `ContainingType.TypeKind`, we already know that `Symbol` is a `DelegateInvoke`, so its container can't be anything else then a delegate, can it?

Reply:
The `Symbol` itself is implicitly declared, but the same also holds for the `Invoke` method call symbol "ordinary" lambdas. I tried to be as narrow and conservative as possible in terms of detecting when we deal with a compiler generated delegate type.
Good point about the typekind check. I will fix that!